utils/codeExecutor.js

```diff
@@ -0,0 +1,34 @@
+const { validateCodeSchema } = require('../schemas/codeSchemas')
+const paiza_io = require("paiza-io")
+
+const execute = (data) => {
+    return new Promise((resolve, reject) => {
+        _execute(data, (successResponse) => resolve(successResponse), (errorResponse) => reject(errorResponse))
+    })
+}
+
+const _execute = (data, successCallback, errorCallback) => {
+
+    if(!validateCodeSchema(data)) {
+        throw Error("Code schema is wrong!")
+    }
+
+    paiza_io(data.language, data.code, data.input, (err, result) => {
+        if(err) {
+            errorCallback(err.message)
+        }
+        successCallback({
+            err: result.stderr,
+            result: result.stdout.trim(),
+            passed: result.stdout.trim() == data.expectedResult.trim()
+        })
+    })
+}
+
+const codeExecutor = {
+    execute
+}
+
+module.exports = {
+    codeExecutor
```

Comment:
Nie wiem czy nie dałbym tutaj po prostu funkcji execute, ale chyba że zostawione pod dalszy rozwój. Plus kwestia z tymi dużymi literami modułów o ile uważasz ją za słuszną.

Reply:
Ogólnie gdybym exportował cały moduł, to się zgadzam powinno być dużą. Ale tutaj eksportuję poszczególne elementy, więc brałem przykład z `useState` i `useEffect` z reacta - też są tak właśnie exportowane i importowane

Reply:
Znaczy codeExecutor nie jest chyba funkcją bo ma w sobie funkcje execute więc bardziej mi osobiście kojarzy z modułem skoro ma w sobie funkcje execute

Reply:
Też mi się wydaje, że przy bezpośrednim exporcie funkcji mała litera jest zasadna, ale execute jest opakowana w moduł codeExecutor, więc chyba codeExecutor powinno być z dużej